### PR TITLE
[ticket/15926] Deny 3.2.x installation if PHP >= 7.3-dev

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-15926

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15926

### CODE REVIEW

File: phpBB/install/app.php [hunk 1]
```diff
@@ -20,9 +20,9 @@
 $phpbb_root_path = '../';
 $phpEx = substr(strrchr(__FILE__, '.'), 1);
 
-if (version_compare(PHP_VERSION, '5.4') < 0)
+if ( version_compare(PHP_VERSION, '5.4.7', '<') || version_compare(PHP_VERSION, '7.3-dev', '>=') )
```
Comment (marc1706) [inline phpBB/install/app.php]: Should be this according to coding guidelines:
```suggestion
if (version_compare(PHP_VERSION, '5.4.7', '<') || version_compare(PHP_VERSION, '7.3-dev', '>='))
```

Comment (3D-I) [reply phpBB/install/app.php]: Ok, though I think that the code's readability gains here.

File: phpBB/install/app.php [hunk 2]
```diff
@@ -20,9 +20,9 @@
 $phpbb_root_path = '../';
 $phpEx = substr(strrchr(__FILE__, '.'), 1);
 
-if (version_compare(PHP_VERSION, '5.4') < 0)
+if (version_compare(PHP_VERSION, '5.4.7', '<') || version_compare(PHP_VERSION, '7.3-dev', '>='))
 {
-	die('You are running an unsupported PHP version. Please upgrade to PHP 5.4.0 or higher before trying to install or update to phpBB 3.2');
+	die('You are running an unsupported PHP version. Please upgrade to PHP equal or greater than 5.4.7 but less than 7.3-dev in order to install or update to phpBB 3.2');
```
Comment (VSEphpbb) [inline phpBB/install/app.php]: should be more like: 
equal to or greater than 5.4.7

**Comment by Solidjeuh** [comment]
Why hardcoded language? Should be added in the language files...

**Comment by DavidIQ** [comment]
Because loading the language would mean loading phpBB, which is what this change attempts to do for PHP 7.3+

**Comment by VSEphpbb** [comment]
Because it has to be installed to load the language sustem, but this error comes before it is installed.

**Pull request merged**